frontend/packages/modelServing/src/components/deployments/DeploymentsTableRow.tsx

```diff
@@ -50,9 +55,28 @@ export const DeploymentRow: React.FC<{
     isModelServingAuthExtension,
     deployment,
   );
+  const [resolvedStartStopActionExtension] = useResolvedDeploymentExtension(
+    isModelServingStartStopAction,
+    deployment,
+  );
```

Comment:
I don't think it's necessary to use the resolved version of this extension hook because this is an async promise that it's returning.

So because resolving it is also a promise, you can just chain the 2 together like so (syntax may be off)
```suggestion
  const tartStopActionExtension = useDeploymentExtension(
    isModelServingStartStopAction,
    deployment,
  );
  
  const onStart = () => resolvedStartStopActionExtension?.properties.patchDeploymentStoppedStatus().then((resolvedFunction) => resolvedFunction(deployment));
```

This way the network call to fetch the function is only done when you are actually using the function, on the onClick / onStart

Reply:
This would also allow you to determine instantly if the StartStopActionExtension exists because it moves to the nonResolved hook

Reply:
I tried this way but couldn't remember the syntax to get the function to resolve. Thanks!